### PR TITLE
update go version to 1.10

### PR DESCRIPTION
**Pull request by andyxning**
This PR will update Golang version to 1.9 as 1.10 has been released. We should use the newer golang version. :)

/cc @brancz

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kube-state-metrics/380)

### CODE REVIEW

**Comment by brancz** [comment]
let's go with 1.10 I would say

**Comment by andyxning** [comment]
@brancz Done. PTAL.

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ sudo: required
 language: go
 
 go:
-- 1.8
+- 1.10
```
Comment (brancz) [inline .travis.yml]: Because yaml, this gets parsed as a decimal and ends up being `1.1`, so we need to quote it.

Comment (andyxning) [reply .travis.yml]: What a big hole it is! Having thought everything in yaml is a string first. After changing it, everything seems ok now.

**Pull request merged**